### PR TITLE
 gnrc_ipv6_nib_6ln: do not mark non-link-local addresses directly VALID

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
The whole address registration looses its point if all addresses are marked valid from the get-go. With this fix addresses are first marked TENTATIVE and only after successful registration marked as VALID.

Additionally because of that, we only have to register addresses that are not VALID yet on reception of router advertisements. This removes the need for the hacky `GNRC_NETIF_FLAGS_6LO_ADDRS_REG` flag that was only introduced to prevent unnecessary re-registration.

Finally, I also piggy-backed a fix for the `ifconfig` shell command, to show TENTATIVE addresses as such. They were not shown as such since sometime during the development of `gnrc_netif` TENTATIVE moved from just being a single flag to being a bit field (to count retries during duplicate address detection). The `ifconfig` command wasn't adapted for that.

### Issues/PRs references
None

### CODE REVIEW

File: sys/net/gnrc/network_layer/ipv6/nib/nib.c [hunk 1]
```diff
@@ -1322,24 +1321,15 @@ static void _auto_configure_addr(gnrc_netif_t *netif, const ipv6_addr_t *pfx,
 #endif  /* GNRC_IPV6_NIB_CONF_6LN */
     }
 
-#if GNRC_IPV6_NIB_CONF_6LN
-    if (gnrc_netif_is_6ln(netif)) {
-        /* don't do this beforehand or risk a deadlock:
-         *  * gnrc_netif_ipv6_addr_add_internal() adds VALID (i.e. manually configured
-         *    addresses to the prefix list locking the NIB's mutex which is already
-         *    locked here) */
-        netif->ipv6.addrs_flags[idx] &= ~GNRC_NETIF_IPV6_ADDRS_FLAGS_STATE_MASK;
```
Comment (miri64) [inline sys/net/gnrc/network_layer/ipv6/nib/nib.c]: FYI: the address flags are initialized TENTATIVE, due to [this line](https://github.com/RIOT-OS/RIOT/pull/8563/files#diff-2b29f6c8fc858131a3d21e4375f5681fR1302).

**Comment by miri64** [comment]
Maybe @bergzand can have a look as well.

**Comment by smlng** [comment]
and after receiving an RA with a PIO (and auto conf flag set) RIOT configures a (global) with that prefix but does not send an NS with that source address and with an ARO, which would be the correct behaviour according to RFC 6775.

**Comment by miri64** [comment]
You are right. I wasn't sure about the link-local behavior and could not found the reference, so I made it for both. Will fix.

**Comment by miri64** [comment]
(note that for a non-routing 6LN to find a router without a border router present, you need #8568.

**Comment by miri64** [comment]
Fixed and adapted 6LN tests.

**Comment by miri64** [comment]
Changed the message of the squash commit, so that I am reminded how the message changed ;-).

**Comment by miri64** [comment]
@smlng ping?

**Comment by smlng** [comment]
looks better now, link-local addresses are valid from start, global or ULA are marked tentative.

**Comment by smlng** [comment]
please squash

**Comment by miri64** [comment]
Squashed.

**Pull request merged**